src/cunumeric/index/advanced_indexing.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "cunumeric/cunumeric.h"
+
+namespace cunumeric {
+
+struct AdvancedIndexingArgs {
```

Comment:
@ipdemes : I thought we have decided to use "nonzero" in this case (when tuple of arrays is passed as indices)

Reply:
I'm not talking about a tuple of arrays in this case, but about the special case of one boolean index array with fewer dimensions than the base array. I thought this case was going to be handled in the future through multi-dim output requirements, and not nonzero. I remember that anything more complex than that (e.g. two boolean indices, or one integer index and one boolean index) was indeed going to be handled with nonzero.

The current implementation does not support different dimensionalities on the two arrays, but it looks like you were building the groundwork for this feature, by having two separate DIM values for the two arrays. If this is not the case, and you only expect to handle equal dimensionalities on the two arrays, then there is no need to pass two dimensions to the C++ side.

If you indeed expect to handle this case in the future, you will also need to know where the boolean index arrays appears within the indexing expression.

Reply:
in the current implementation, `[:,b1]` will become a `tuple (Slice(None), b1)` -> `nonzero` will be called for b1. I think I can implement `smart` logic for tuples so it will handle this case with `AdvancedIndexing` task, but, maybe, it should be part of the separate PR ?

Reply:
Yes, let's leave it for the future PR